vip-scanner/checks/VIPRestrictedPatternsCheck.php

```diff
@@ -6,30 +6,116 @@ function check( $files ) {
+			),
+			'missing-dotcom-from-vip' => array(
+				'expression' => '/WordPress VIP/msiU',
+				'level'      => 'Warning',
+				'note'       => 'Please use "WordPress.com VIP" rather than "WordPress VIP"',
+			),
+			'database-table-alteration' => array(
+				'expression' => '/(\$wpdb->|mysql_)+.+(ALTER)+\s+/msiU',
+				'level'      => 'Blocker',
+				'note'       => 'Possible database table alteration',
+			),
+			'database-table-creation' => array(
+				'expression' => '/(\$wpdb->|mysql_)+.+(CREATE)+\s+/msiU',
+				'level'      => 'Blocker',
+				'note'       => 'Possible database table creation',
+			),
+			'database-table-deletion' => array(
+				'expression' => '/(\$wpdb->|mysql_)+.+(DROP)+\s+/msiU',
+				'level'      => 'Blocker',
+				'note'       => 'Possible database table deletion',
+			),
+			'database-delete-query' => array(
+				'expression' => '/(\$wpdb->|mysql_)+.+(DELETE)+\s+(FROM)+\s+/msiU',
+				'level'      => 'Warning',
+				'note'       => 'Direct database delete query',
+			),
+			'database-select-query' => array(
+				'expression' => '/(\$wpdb->|mysql_)+.+(SELECT)+\s.+/msiU',
+				'level'      => 'Note',
+				'note'       => 'Direct Database select query',
+			),
+			'direct-database-query' => array(
+				'expression' => '/(^GLOBAL)(\$wpdb->|mysql_)+/msiU',
+				'level'      => 'Warning',
+				'note'       => 'Possible direct database query',
+			),
+			'output-of-restricted-variables' => array(
+				'expression' => '/(echo|\<\?\=)+(?!\s+\(?\s*(?:isset|typeof)\(\s*)[^;]+(\$GLOBALS|\$_SERVER|\$_GET|\$_POST|\$_REQUEST)+/msiU',
+				'level'      => 'Warning',
+				'note'       => 'Possible output of restricted variables',
+			),
+			'working-with-superglobals' => array(
+				'expression' => '/(\$GLOBALS|\$_SERVER|\$_GET|\$_POST|\$_REQUEST)+/msiU',
+				'level'      => 'Note',
+				'note'       => 'Working with superglobals',
+			),
+			'non-whitelisted-server-superglobal' => array(
+				'expression' => '/(\$_SERVER\[(?!(\'|"REQUEST_URI|SCRIPT_FILENAME|HTTP_HOST\'|"))([^]]+|)\])+/msiU',
+				'level'      => 'Blocker',
+				'note'       => 'Non whitelisted $_SERVER superglobals found in this file',
+			),
+			'unsafe-pre_option-hook-use' => array(
+				'expression' => '/(pre_)?option_(blogname|siteurl|post_count)/msiU',
+				'level'      => 'Blocker',
+				'note'       => 'possible unsafe use of pre_option_* hook',
+			),
+			'direct-query_vars-access' => array(
+				'expression' => '/\$wp_query->query_vars\[.*?\][^=]*?\;/msi',
+				'level'      => 'Warning',
+				'note'       => 'Possible direct query_vars access, should use get_query_var() function',
+			),
+			'direct-query_vars-modification' => array(
+				'expression' => '/\$wp_query->query_vars\[.*?\]\s*?\=.*?\;/msi',
+				'level'      => 'Warning',
+				'note'       => 'Possible direct query_vars modification, should use set_query_var() function',
+			),
+			'variable-variables' => array(
+				/*
+					This regex is a bit complicated, so I will break it down:
```

Comment:
Would it make sense to break this regex up into a few, more manageable regexes that each search for specific patterns?